lib/3scale/backend/usage_limit.rb

```diff
@@ -22,20 +22,15 @@ class << self
 
         def load_all(service_id, plan_id)
           metric_ids = Metric.load_all_ids(service_id)
-          return metric_ids if metric_ids.empty?
-
-          results = []
-          with_pairs_and_values service_id, plan_id, metric_ids do |pair, value|
-            value and results << new(service_id: service_id,
-                                     plan_id: plan_id,
-                                     metric_id: pair[0],
-                                     period: pair[1],
-                                     value: value.to_i)
-          end
-          results
+          generate_for_metrics(service_id, plan_id, metric_ids)
```

Comment:
use load for metrics and implement there the body of generate for metrics? and maybe sort these so that if any request comes with all metrics for a service and no body can also benefit from memoization?

Reply:
I thought about that. The problem is that some tests that call `load_all` rely on clearing the cache when a limit is added or deleted: https://github.com/3scale/apisonator/blob/d51ff8deadd053f9b1eeaa75e5c80ba19d265a7e/lib/3scale/backend/usage_limit.rb#L108

We can't include the new method there because it also accepts `metrics_ids` as a param. We want it to be memoized, but we can't invalidate it when a usage limit is added/deleted.

I think the tests shouldn't rely on the memoizer because in practice, there will be many processes and it will only be cleared in the one that receives the request to add/delete a usage limit. However, in this case, I preferred to add a bit of duplication rather than changing the existing behavior and tests. I can change this if you have a clear preference. Personally, I don't think it's too important.

Reply:
Ok, sounds reasonable.